src/main/js/controls/Graph/Graph.js

```diff
@@ -377,6 +381,82 @@ class Graph extends Construct {
         return this;
     }
 
+    /**
+     * Updates the graph axisData and content.
+     *
+     * @returns {Graph} - Graph instance
+     * @param {Array} graphData - Input array that holds updated values and key
+     */
+    reflow(graphData) {
+        this.config.axis.x.domain = getDomain(
+            this.config.axis.x.type,
+            this.config.axis.x.lowerLimit,
+            this.config.axis.x.upperLimit
+        );
+        const width = getXAxisWidth(this.config);
+        scaleGraph(this.scale, this.config);
+        const axisData = d3.svg
+            .axis()
+            .scale(this.scale.x)
+            .ticks(
+                Math.max(
+                    Math.ceil(width / constants.MAX_TICK_VARIANCE),
+                    constants.MIN_TICKS
+                )
+            )
+            .orient(this.config.axis.x.orientation);
+
+        const svg = d3
+            .selectAll("svg")
+            .selectAll(`.${styles.axis}.${styles.axisX}`)
+            .data(this.config.axis.x.domain);
+        svg.enter();
+        svg.transition()
+            .attr("class", styles.axis)
+            .attr("class", styles.axisX)
+            .attr("aria-hidden", !this.config.axis.x.show)
+            .attr(
+                "transform",
+                `translate(${getXAxisXPosition(
+                    this.config
+                )}, ${getXAxisYPosition(this.config)})`
+            )
+            .call(axisData);
+        svg.exit().remove();
+
+        if (graphData && this.contentKeys.includes(graphData.key)) {
+            let position;
+            this.contentConfig.forEach((config, index) => {
+                if (config.key === graphData.key) position = index;
+            });
+            this.contentConfig[position].values = graphData.values;
+            // if (utils.notEmpty(this.content[position].config.values)) {
+            //     removeNoDataView(this.svg);
+            // } else {
+            //     drawNoDataView(this.config, this.svg)
+            // }
+            scaleGraph(this.scale, this.config);
+            this.content[position].update(this, graphData);
```

Comment:
This is good, update only contents that needs reflow. I think here you can instruct the respective content to update their state, instead of doing line 432.

Reply:
Did you mean something like this [line 192 Line.js](https://github.com/cerner/carbon-graphs/pull/138/files#diff-1c4c0200889225a0e2bc46a0e62648ccR192)? @abhijit945